src/libmongoc/tests/test-mongoc-transactions.c

```diff
@@ -1166,6 +1166,39 @@ test_max_commit_time_ms_is_reset (void *ctx)
    mock_rs_destroy (rs);
 }
 
+void
+test_snapshot_session_prose_1 (void *ctx)
+{
+   mongoc_client_t *client = NULL;
+   mongoc_session_opt_t *session_opts = NULL;
+   mongoc_uri_t *uri = NULL;
+   bson_error_t error;
+   bool r;
+
+   uri = test_framework_get_uri ();
+   client = test_framework_client_new_from_uri (uri, NULL);
+   BSON_ASSERT (client);
+
+   session_opts = mongoc_session_opts_new ();
+   mongoc_session_opts_set_causal_consistency (session_opts, true);
+   mongoc_session_opts_set_snapshot (session_opts, true);
+
+   /* assert that starting session with causal consistency and snapshot enabled
+    * results in an error. */
+   r = mongoc_client_start_session (client, session_opts, &error);
+   ASSERT (!r);
+   ASSERT (error.domain == MONGOC_ERROR_CLIENT);
```

Comment:
Minor: the following can be simplified to:

```suggestion
   ASSERT_ERROR_CONTAINS (error,
                          MONGOC_ERROR_CLIENT, 
                          MONGOC_ERROR_CLIENT_SESSION_FAILURE,
                          "Only one of causal consistency and snapshot can be enabled.");
```